### DISCUSSION TITLE
Not able to write test for "toBrowser"

### DISCUSSION THREAD

**Discussion by ghost** [spatie/simple-excel#68] [Q&A]
We are streaming the file on browser with the following code.

SimpleExcelWriter::streamDownload('commitments.csv')
            ->noHeaderRow()
            ->addRow($header_row)
            ->addRows($commitment_arr)
            ->toBrowser();

But not able to write test for it as, it gives CSV file output on console itself.
regardless of whatever function we use after hitting the end point in test.

We had tried with streamDownload, assertDownload.

**Comment by nunodonato** [comment]
Same problem here. @mr-yogesh-galav did you find a solution?

**Reply by 77media-creations** [reply]
Same issue here. I'd love to see if someone has a solution.

**Comment by tylerwiegand** [comment]
bumpin the issue

is there a way to achieve this without rewriting the package? or do we basically have to create the file somewhere and serve it up as a "laravel" file download?
